From 6b7fcda3509f64a0dc93cdb31f8f490dc8d2c94d Mon Sep 17 00:00:00 2001
From: Tianshu Qiu <tian.shu.qiu@intel.com>
Date: Wed, 29 Apr 2015 17:11:40 +0800
Subject: [PATCH 375/462] pixter: Add metadata support

Change-Id: Ic1eb6ec6f93661908e3c2c27d498ca73eebdd1f4
Fixes-Issue: #JIMINAN-34539
Signed-off-by: Tianshu Qiu <tian.shu.qiu@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/53529
Approver: Bin B Han <bin.b.han@intel.com>
Reviewed-by: Bin B Han <bin.b.han@intel.com>
Validation-Android: Wei Tao <wei.tao@intel.com>
Integrator: Wei Tao <wei.tao@intel.com>
Build: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Maintainer: WeiX Meng <weix.meng@intel.com>
---
 .../external_drivers/camera/drivers/media/i2c/pixter.c   | 16 ++++++++++++++++
 .../external_drivers/camera/drivers/media/i2c/pixter.h   |  6 ++++++
 2 files changed, 22 insertions(+)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/pixter.c b/drivers/external_drivers/camera/drivers/media/i2c/pixter.c
index c6dcad0..39d9528 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/pixter.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/pixter.c
@@ -68,6 +68,9 @@ static struct regmap_config pixter_reg_config = {
 	.val_format_endian = REGMAP_ENDIAN_NATIVE,
 };
 
+static const uint32_t pixter_embedded_effective_size[2]	= {
+	PIXTER_METADATA_EFFECTIVE_SIZE0, PIXTER_METADATA_EFFECTIVE_SIZE1};
+
 static struct pixter_format_bridge format_bridge[] = {
 	{"", 0, ATOMISP_INPUT_FORMAT_BINARY_8, 8},
 	{"RGGB10", V4L2_MBUS_FMT_SRGGB10_1X10, ATOMISP_INPUT_FORMAT_RAW_10, 10},
@@ -653,6 +656,19 @@ static int pixter_s_mbus_fmt(struct v4l2_subdev *sd,
 		stream_info->ch_id = stream_info->stream;
 	dev->vc_setting[stream_info->ch_id] =
 		dev->settings[dev->cur_setting].vc[stream_info->ch_id];
+
+#if PIXTER_METADATA_EN
+	dev->mipi_info->metadata_width =
+	    (dev->settings[dev->cur_setting].vc[stream_info->ch_id].width *
+	    format_bridge[dev->settings[dev->cur_setting].vc[stream_info->ch_id].format].bpp + 7) / 8;
+	dev->mipi_info->metadata_height = PIXTER_METADATA_LINES;
+	dev->mipi_info->metadata_effective_width = pixter_embedded_effective_size;
+	dev->mipi_info->metadata_format = PIXTER_METADATA_FORMAT;
+	dev_dbg(&client->dev, "Metadata: %dx%d fmt=%d\n",
+		dev->mipi_info->metadata_width,
+		dev->mipi_info->metadata_height,
+		dev->mipi_info->metadata_format);
+#endif
 	mutex_unlock(&dev->input_lock);
 	dev_dbg(&client->dev, "%s w:%d h:%d code: 0x%x stream: %d\n", __func__,
 			fmt->width, fmt->height, fmt->code,
diff --git a/drivers/external_drivers/camera/drivers/media/i2c/pixter.h b/drivers/external_drivers/camera/drivers/media/i2c/pixter.h
index 64fcd2a..3004246 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/pixter.h
+++ b/drivers/external_drivers/camera/drivers/media/i2c/pixter.h
@@ -74,6 +74,12 @@
 #define PIXTER_RONLY		S_IRUSR
 #define PIXTER_RW		(S_IRUSR | S_IWUSR)
 
+#define PIXTER_METADATA_EN	0
+#define PIXTER_METADATA_LINES	2
+#define PIXTER_METADATA_EFFECTIVE_SIZE0	100
+#define PIXTER_METADATA_EFFECTIVE_SIZE1	100
+#define PIXTER_METADATA_FORMAT	ATOMISP_INPUT_FORMAT_EMBEDDED
+
 #ifdef CONFIG_GMIN_INTEL_MID
 /* Supported TYPE:
 	RAW_CAMERA
-- 
1.9.1

